inject.js

```diff
@@ -1,65 +1,71 @@
 const nest = require('depnest');
-const index = require('./index');
-const {
-  h
-} = require('mutant');
+const { h, onceTrue } = require('mutant');
+// const { isMsg } = require('ssb-ref')
 
-var onceTrue = require('mutant/once-true');
+const index = require('./index');
 
 exports.gives = nest({
-  'app.html': {
-    page: true,
-    menuItem: true
-  }
+  'app.html.menuItem': true, 
+  'app.page.chessIndex': true,
+  'router.sync.routes': true,
 })
 
 exports.needs = nest({
+  'app.page.chessIndex': 'first',
   'sbot.obs.connection': 'first',
   'backlinks.obs.for': 'first'
 });
 
 exports.create = function(api) {
-  const route = '/chess'
-
-  const topLevelDomElement = document.createElement('div');
-  topLevelDomElement.className = "ssb-chess-container";
-
+  const topLevelDomElement = h('div.ssb-chess-container', { title: '/chess' })
   var pageLoaded = false;
 
   return nest({
-    'app.html': {
-      menuItem: menuItem,
-      page: chessPage
-    }
+    'app.html.menuItem': menuItem, 
+    'app.page.chessIndex': chessIndex,
+    'router.sync.routes': routes,
   })
 
   function menuItem(handleClick) {
     return h('a', {
       style: {
         order: 0
       },
-      'ev-click': () => handleClick(route)
-    }, route)
+      'ev-click': () => handleClick({ page: 'chess' })
+    }, '/chess')
   }
 
-  function chessPage(path) {
-    if (path !== route) {
-      return;
-    } else if (pageLoaded) {
+  function chessIndex(location) {
+    if (pageLoaded) {
       return topLevelDomElement;
-    } else {
-      onceTrue(api.sbot.obs.connection(), (sbot) => {
+    } 
+    
+    onceTrue(api.sbot.obs.connection(), (sbot) => {
 
-        var injectedApi = {
-          backlinks: api.backlinks.obs.for
-        }
+      var injectedApi = {
+        backlinks: api.backlinks.obs.for
+      }
 
-        index(topLevelDomElement, sbot, injectedApi);
-        pageLoaded = true;
+      index(topLevelDomElement, sbot, injectedApi);
+      pageLoaded = true;
+    });
 
-      });
+    return topLevelDomElement;
+  }
 
-      return topLevelDomElement;
-    }
+  function routes (sofar = []) {
+    const pages = api.app.page
+
+    // loc = location, an object with all the info required to specify a location
+    const routes = [
+//    [ loc => loc.page === 'chessGame' && isMsg(loc.game), pages.chessGame ], // example
+//    [ loc => loc.page === 'chess' && isMsg(loc.game), pages.chessGame ],     // alternative
+
+      [ loc => loc.page === 'chess', pages.chessIndex ],
+    ]
```

Comment:
How the router works:

In **Patchcore** has `router/sync/router.js` which reduces (collects and compiles) every given `router.sync.routes`. So here we're contributing to that.

A 'routes' is expected to return an Array of routes. Each route is of the format: 
`[ validationFunction,  actionFunction ]`

When the router is given a `location` (ideally an object) it look through the routes (in order) until it hits the first route where `validationFunction(location) == truthy`, the router then returns `actionFunction(location)`.

`actionFunction` can do whatever - it might be a chunk of html, or it might be some action and return nothing.

In **Patchbay** the thing which interacts with this is `app/sync/goTo.js`. I'd like to refactor this area in the future. `goTo` ... (via some stuff) calls api.router.sync.router(location) .. and **if this returns anything** (like html) it shoves it in a tab (if a tab matching that location don't already exist).
I mention all this to point out that the router can be used to render pages OR do work for you in some other way.